velox/expression/CastExpr.cpp

```diff
@@ -760,6 +789,8 @@ void CastExpr::applyPeeled(
         "Cast from {} to {} is not supported",
         fromType->toString(),
         toType->toString());
+  } else if (fromType->isIntervalYearMonth() && toType->kind() == TypeKind::INTEGER) {
```

Comment:
This logic is used by both Presto and Spark, would you check if Presto's behaviour is aligned?